logdetective/server/server.py

```diff
@@ -600,20 +599,30 @@ async def check_artifacts_file_size(job):
 
 
 async def comment_on_mr(
-    merge_request_id: int,
+    project: gitlab.v4.objects.Project,
+    merge_request_iid: int,
     job: gitlab.v4.objects.ProjectJob,
     log_url: str,
     response: StagedResponse,
 ):
     """Add the Log Detective response as a comment to the merge request"""
     LOG.debug(
-        "Primary Explanation for MR %d: %s", merge_request_id, response.explanation.text
+        "Primary Explanation for %s MR %d: %s",
+        project.name,
+        merge_request_iid,
+        response.explanation.text,
     )
 
     # Get the formatted comment.
     comment = await generate_mr_comment(job, log_url, response)
 
+    # Look up the merge request
+    merge_request = await asyncio.to_thread(
+        project.mergerequests.get, merge_request_iid
+    )
+
     # Submit a new comment to the Merge Request using the Gitlab API
+    await asyncio.to_thread(merge_request.discussions.create, {"body": comment})
```

Comment:
I imagine that the `python-gitlab` objects don't support asynchronous methods, which is unfortunate. In my experience, working with threads in an asynchronous application doesn't scale well. The `ThreadPoolExecutor` typically has only a limited number of workers, which can become a bottleneck.

Reply:
Yeah, it's unfortunate, but I think we'll be alright in this case, as each of these tasks are relatively short and self-contained. If we were trying to run long-term tasks in the threads, I'd definitely find a new approach.

The obvious alternative would be for us to essentially reimplement the python-gitlab library and build it atop the `aiohttp` or `httpx` module, as is proposed in https://github.com/python-gitlab/python-gitlab/issues/1025 but I don't think we realistically have the time to do that right now.

We could also look at switching over to the GraphQL API for Gitlab and using `aiographql`, but that is a significant change in the API behavior.

I'll open a ticket to investigate whether either of those strategies would be worth the time to implement.

Reply:
I think that creating a ticket for investigate this **in future** is perfect! I agree that right now this is not a problem at all :) I was just a bit worried for the future. Thanks.